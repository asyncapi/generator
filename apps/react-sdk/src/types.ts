import React from "react";
import { AsyncAPIDocument } from "@asyncapi/parser";

export type PropsWithChildrenContent<P> = P & {
  childrenContent?: string;
}

export type FC<P = {}> = FunctionComponent<P>;
export type FunctionComponent<P = {}> = React.FunctionComponent<PropsWithChildrenContent<P>>;
export class Component<P = {}> extends React.Component<PropsWithChildrenContent<P>> {}

/**
 * Shape of the context passed to template
 */
export interface TemplateContext<P = Record<string, any>>{
  asyncapi: AsyncAPIDocument,
  params: P,
  originalAsyncAPI: string,
  [key: string]: any,
}

/**
 * Signature for template function
 */
export type TemplateFunction<R = React.ReactElement | undefined> = (context: TemplateContext) => R | Promise<R>;

/**
 * Options for transpiling files.
 */
export type TranspileFilesOptions = {
  /**
   * Should all files in a directory including those in subdirectories be included
   */
  recursive?: boolean
}

export type TemplateRenderMetadata = {
  fileName?: string;
  permissions?: string;
}

export type TemplateRenderResult = {
  metadata: TemplateRenderMetadata;
  content: string;
}