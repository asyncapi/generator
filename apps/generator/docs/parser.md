---
title: "Parser"
weight: 60
---

## Parser

The **AsyncAPI Parser** is a powerful library for parsing and validating [AsyncAPI documents](asyncapi-document) in both Node.js and browser environments. It supports YAML and JSON inputs and ensures correctness through schema validation.

### âœ… Supported schema formats:

- AsyncAPI (v2 and v3)
- OpenAPI
- JSON Schema
- Avro
- RAML

These schemas are automatically transformed into JSON Schema format internally, simplifying access and manipulation.

---

## Key Features

- **Parsing** of AsyncAPI YAML/JSON files.
- **Validation** against AsyncAPI and referenced schemas.
- **Dereferencing** of `$ref` values into inlined content.
- **Helper functions** to access channels, messages, schemas, and more.

---

## Example: AsyncAPI v3 Document

Below is a minimal v3.0.0-compliant AsyncAPI document using `channels`, `operations`, and component references:

```yaml
asyncapi: 3.0.0
info:
  title: Comments Service
  version: 1.0.0
  description: Handles all comment-related events.
servers:
  dev:
    host: test.mosquitto.org
    protocol: mqtt
channels:
  sendCommentLiked:
    address: comment/liked
    messages:
      commentLiked:
        $ref: '#/components/messages/commentLiked'
  sendCommentUnliked:
    address: comment/unliked
    messages:
      commentUnliked:
        $ref: '#/components/messages/commentUnliked'
operations:
  sendCommentLiked:
    action: send
    channel:
      $ref: '#/channels/sendCommentLiked'
  sendCommentUnliked:
    action: send
    channel:
      $ref: '#/channels/sendCommentUnliked'
components:
  messages:
    commentLiked:
      payload:
        $ref: '#/components/schemas/commentReaction'
    commentUnliked:
      payload:
        $ref: '#/components/schemas/commentReaction'
  schemas:
    commentReaction:
      type: object
      properties:
        commentId:
          type: string
