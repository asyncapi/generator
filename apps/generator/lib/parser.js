const fs = require('fs');
const { convertToOldAPI } = require('@asyncapi/parser');
const { ConvertDocumentParserAPIVersion, NewParser } = require('@smoya/multi-parser');

const parser = module.exports;

/**
 * Convert the template defined value `apiVersion: 'v1'` to only contain the numeric value `1`.
 */
parser.sanitizeTemplateApiVersion = (apiVersion) => {
  if (!apiVersion) return;
  if (apiVersion && apiVersion.length > 1) {
    return Number(apiVersion.substring(1));
  }
  return Number(apiVersion);
};

parser.parse = async (asyncapi, oldOptions, generator) => {
  let apiVersion = this.sanitizeTemplateApiVersion(generator.templateConfig.apiVersion);
  // Defaulting to apiVersion v1 to convert it to the Parser-API v1 afterwards.
  if (!this.usesNewAPI(generator.templateConfig)) {
    apiVersion = 1;
  }
  const options = convertOldOptionsToNew(oldOptions, generator);
  const parser = NewParser(apiVersion, {parserOptions: options, includeSchemaParsers: true});
  const { document, diagnostics } = await parser.parse(asyncapi, options);
  if (!document) {
    return {document, diagnostics};
  }
  const correctDocument = this.getProperApiDocument(document, generator.templateConfig);
  return {document: correctDocument, diagnostics};
};

/**
 * If the template expect one of the Parser-API versions, it must be above 0
 */
parser.usesNewAPI = (templateConfig = {}) => {
  return this.sanitizeTemplateApiVersion(templateConfig.apiVersion) > 0;
};

/**
 * Based on the current parsed AsyncAPI document, convert it to expected API version from the template.
 */
parser.getProperApiDocument = (asyncapiDocument, templateConfig = {}) => {
  const apiVersion = this.sanitizeTemplateApiVersion(templateConfig.apiVersion);
  if (apiVersion === undefined) {
    // Convert to old API from JS Parser v1
    return convertToOldAPI(asyncapiDocument);
  }
  return ConvertDocumentParserAPIVersion(asyncapiDocument, apiVersion);
};

// The new options for the v2 Parser are different from those for the v1 version, but in order not to release Generator v2, we are converting the old options of Parser to the new ones.
function convertOldOptionsToNew(oldOptions, generator) {
  if (!oldOptions) return;
  const newOptions = {};

  if (typeof oldOptions.path === 'string') {
    newOptions.source = oldOptions.path;
  }
  if (typeof oldOptions.applyTraits === 'boolean') {
    newOptions.applyTraits = oldOptions.applyTraits;
  }

  const resolvers = [];
  if (generator && generator.mapBaseUrlToFolder && generator.mapBaseUrlToFolder.url) {
    resolvers.push(...getMapBaseUrlToFolderResolvers(generator.mapBaseUrlToFolder));
  }
  if (oldOptions.resolve) {
    resolvers.push(...convertOldResolvers(oldOptions.resolve));
  }

  if (resolvers.length) {
    newOptions.__unstable = {};
    newOptions.__unstable.resolver = {
      resolvers,
    };
  }

  return newOptions;
}

/**
 * Creates a custom resolver that maps urlToFolder.url to urlToFolder.folder
 * Building your custom resolver is explained here: https://apitools.dev/json-schema-ref-parser/docs/plugins/resolvers.html
 *
 * @private
 * @param  {object} urlToFolder to resolve url e.g. https://schema.example.com/crm/ to a folder e.g. ./test/docs/.
 * @return {{read(*, *, *): Promise<unknown>, canRead(*): boolean, order: number}}
 */
function getMapBaseUrlToFolderResolvers({ url: baseUrl, folder: baseDir }) {
  const resolver = {
    order: 1,
    canRead: true,
    read(uri) {
      return new Promise(((resolve, reject) => {
        const path = uri.toString();
        const localpath = path.replace(baseUrl, baseDir);
        try {
          fs.readFile(localpath, (err, data) => {
            if (err) {
              reject(`Error opening file "${localpath}"`);
            } else {
              resolve(data.toString());
            }
          });
        } catch (err) {
          reject(`Error opening file "${localpath}"`);
        }
      }));
    }
  };

  return [
    { schema: 'http', ...resolver, },
    { schema: 'https', ...resolver, },
  ];
};

function convertOldResolvers(resolvers = {}) { // NOSONAR
  if (Object.keys(resolvers).length === 0) return [];

  return Object.entries(resolvers).map(([protocol, resolver]) => {
    return {
      schema: protocol,
      order: resolver.order || 1,
      canRead: (uri) => canReadFn(uri, resolver.canRead),
      read: (uri) => {
        return resolver.read({ url: uri.valueOf(), extension: uri.suffix() });
      }
    };
  });
}

function canReadFn(uri, canRead) {
  const value = uri.valueOf();
  if (typeof canRead === 'boolean') return canRead;
  if (typeof canRead === 'string') return canRead === value;
  if (Array.isArray(canRead)) return canRead.includes(value);
  if (canRead instanceof RegExp) return canRead.test(value);
  if (typeof canRead === 'function') {
    return canRead({ url: value, extension: uri.suffix() });
  }
  return false;
}
