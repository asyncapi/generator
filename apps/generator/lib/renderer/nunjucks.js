const Nunjucks = require('nunjucks');
const nunjucksExport = module.exports;

/**
 * Configures Nunjucks templating system
 * 
 * @private
 * @param {boolean} debug flag
 * @param {string} templateDir path
 */
nunjucksExport.configureNunjucks = (debug, templateDir) => {
  const config = {};
  if (debug) config.dev = true;

  return new Nunjucks.Environment(new Nunjucks.FileSystemLoader(templateDir), config);
};

/**
 * Renders the template with nunjucks and returns a string.
 * 
 * @param {import('@asyncapi/parser').AsyncAPIDocument} asyncapiDocument 
 * @param {string} templateString template filecontent to be rendered with nunjucks
 * @param {string} filePath path to the template file
 * @param {Object} extraTemplateData Extra data to pass to the template.
 * @param {Object} templateParams provided template parameters
 * @param {string} originalAsyncAPI 
 * @param {} nunjucks instance
 * @return {Promise}
 */
nunjucksExport.renderNunjucks = (asyncapiDocument, templateString, filePath, extraTemplateData, templateParams, originalAsyncAPI, nunjucks) => {
  return new Promise((resolve, reject) => {
    nunjucks.renderString(templateString, {
      asyncapi: asyncapiDocument,
      params: templateParams,
      originalAsyncAPI,
      ...extraTemplateData
    }, { path: filePath }, (err, result) => {
      if (err) return reject(err);
      resolve(result);
    });
  });
};
