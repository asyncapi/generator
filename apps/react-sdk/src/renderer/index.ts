export * from "./renderer";
export * from "./template";
