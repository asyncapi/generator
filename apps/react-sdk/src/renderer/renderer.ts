import { PropsWithChildrenContent } from "../types";

/**
 * Normalizes given props with render children to string and save value to `childrenContent` prop
 * @private
 * @param {Any} props 
 * @returns {Any} normalized props.
 */
function normalizeProps<P = any>(props: any): PropsWithChildrenContent<P> {
  return { 
    ...props, 
    childrenContent: render(props.children),
  }
}

/**
 * Executes a `render` method on a given component (in the case of a class component)
 * or executes the component itself (a functional component) to get pure string or complex value for the next operations
 * @private
 * @param {React.ReactElement} element a given component or complex form
 * @returns {(React.ReactElement | string)}
 */
function createElement(element: React.ReactElement): React.ReactElement | string {
  if (!element) {
    return "";
  }
  const typeOf = typeof element.type;

  if (typeOf === 'string') {
    // HTML (also not standard) tags case
    throw new Error("HTML tags is not supported yet.");
  } else if (typeOf === 'symbol') {
    // internal React types like Fragments, Portal etc. We skip them.
    return render(element.props.children);
  } else if (typeOf === 'function') {
    // custom components case
    const type = element.type as any;
    const prototype = type.prototype;

    // Class component case
    if (prototype && typeof prototype.isReactComponent === "object") {
      const clazzComp = new type(normalizeProps(element.props));
      return createElement(clazzComp.render());
    }
    // Function component case
    return createElement(type(normalizeProps(element.props)));
  }

  return element || "";
}

/**
 * Renders given component to string
 * 
 * @param {ReactNode} component a given component to rendering
 * @example
 * function Component({ textProp }) {
 *   return <>{textProp}</>
 * }
 * render(<Component textProp="someText" />)
 * @returns {string}
 */
export function render(component: React.ReactNode): string {
  let content = "";
  const typeOf = typeof component;
  if (typeOf === 'string') {
    content += component;
  } else if (Array.isArray(component)) {
    content += component.map(child => {
      const childValue = createElement(child as React.ReactElement);
      return render(childValue);
    }).join("");
  } else if (typeOf === "object") {
    content += createElement(component as React.ReactElement);
  }
  return content;
}
