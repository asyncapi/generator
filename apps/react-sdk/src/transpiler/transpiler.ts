import Path from 'path';

import { rollup } from 'rollup';
import babel from '@rollup/plugin-babel';

import { getStatsInDir } from '../utils';
import { TranspileFilesOptions } from '../types';

const ROOT_DIR = Path.resolve(__dirname, '../..');

/**
 * Transpile files in a given directory (and sub directory if recursive option are passed) and write it to an output directory, if no errors are thrown it completed successfully.
 * 
 * @param directory to transpile.
 * @param outputDir to write the transpiled files to.
 * @param options any extra options that should be passed.
 */
export async function transpileFiles(directory: string, outputDir: string, options?: TranspileFilesOptions) {
    const { files, dirs } = await getStatsInDir(directory);
    if (files.length) {
        /**
         * WHEN ADDING PLUGINS to transform the input keep in mind that 
         * IF any of these changes the actual original content in some way
         * the output is not able to produce accurate source map to the original file.
         * 
         * An example of this is using the `sourceMaps: 'inline'` configuration for the babel plugin.
         */
        const bundles = await rollup({
            input: files,
            onwarn: () => {},
            plugins: [
                babel({
                    cwd: ROOT_DIR,
                    babelHelpers: "bundled",
                    plugins: [
                        "source-map-support",
                    ],
                    presets: [
                        ["@babel/preset-env", {
                            targets: { node: "12.16" },
                        }],
                        ["@babel/preset-react", {
                            runtime: "automatic",
                        }],
                    ],
                })
            ]
        })
        await bundles.write({
            format: "commonjs",
            sourcemap: true,
            dir: outputDir,
            exports: "auto",
            paths: {
              'react/jsx-runtime': 'react/cjs/react-jsx-runtime.production.min',
            }
        })
    }

    // Check if we should transpile all subdirs
    if (options?.recursive === true && dirs.length > 0) {
        for (const subdir of dirs) {
            const subdirPath = Path.parse(subdir);
            await transpileFiles(subdir, Path.resolve(outputDir, subdirPath.base), options);
        }
    }
}
