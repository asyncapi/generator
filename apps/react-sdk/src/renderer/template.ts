import React from "react";

import { render } from "./renderer";
import { isJsFile } from "../utils";
import { TemplateContext, TemplateFunction, TemplateRenderResult } from "../types";

/**
 * render a file with react. This function automatically transforms jsx to js before importing the component.
 * 
 * @param filepath the path to file to render
 */
export async function renderTemplate(filepath: string, context: TemplateContext): Promise<TemplateRenderResult[] | TemplateRenderResult | undefined> {
  if (!isJsFile(filepath)) {
    return undefined;
  }

  let data = undefined;
  try {
    const component = await importComponent(filepath);
    if (component === undefined) {
      return undefined;
    }
    data = await component(context);
  } catch(err) {
    throw err;
  }

  // undefined, null etc. cases
  if (!data) {
    return undefined;
  }

  if (Array.isArray(data)) {
    return data.map(file => file && renderFile(file)).filter(Boolean);
  }
  return renderFile(data);
}

/**
 * Imports a given file and return the imported component
 * 
 * @private
 * @param filepath to import
 */
function importComponent(filepath: string): Promise<TemplateFunction | undefined> {
  return new Promise((resolve, reject) => {
    try {
      // we should import component only in NodeJS
      if (require === undefined) resolve(undefined);
      // remove from cache imported file
      delete require.cache[require.resolve(filepath)];

      const component = require(filepath);
      if (typeof component === "function") resolve(component);
      if (typeof component.default === "function") resolve(component.default);
      resolve(undefined);
    } catch(err) {
      reject(err);
    }
  });
}

/**
 * Render a single File component.
 * 
 * @private
 * @param {React.ReactElement} file to import
 */
function renderFile(file: React.ReactElement): TemplateRenderResult | undefined {
  if (typeof file !== "object") {
    return undefined;
  }
  const { type, props = {} } = file;

  // if no File component is found as root, don't render it.
  if (typeof type !== "function" || type.name !== "File") {
    return undefined;
  }

  return {
    content: render(props.children),
    metadata: {
      fileName: props.name,
      permissions: props.permissions,
    }
  };
}
