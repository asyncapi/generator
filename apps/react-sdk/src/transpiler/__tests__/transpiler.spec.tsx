import fs from 'fs';
import path from 'path';
import { promisify } from 'util';
import { AsyncAPIDocument } from "@asyncapi/parser";

import { transpileFiles } from "../transpiler";
import { renderTemplate } from "../../renderer";
import { TemplateRenderResult } from "../../types";

const readFile = promisify(fs.readFile);

describe('Transpiler', () => {
  const testFiles = path.resolve(__dirname, './testfiles');
  const outputFiles = path.resolve(__dirname, './__transpiled_testfiles');
  beforeAll(async (done) => {
    try {
      await transpileFiles(testFiles, outputFiles, {
        recursive: true
      });
      done();
    } catch (e) {
      console.log(e);
      done(e);
    }
  });
  describe('should transpile CommonJS files', () => {
    describe('with a simple setup', () => {
      const commonjs_testFile = path.resolve(outputFiles, './CommonJS/simple.js');
      const commonjs_testFileMap = path.resolve(outputFiles, './CommonJS/simple.js.map');
      test('and import correctly', async () => {
        const content = await readFile(commonjs_testFile, 'utf8');
        expect(switchToUnixLinebreaks(content)).toMatchSnapshot();
        const mapContent = await readFile(commonjs_testFileMap, 'utf8');
        expect(switchToUnixLinebreaks(mapContent)).toMatchSnapshot();
        expect(await import(commonjs_testFile)).toBeDefined();
      });
      test('and render correctly', async () => {
        const content = await renderTemplate(commonjs_testFile, { asyncapi: {} as AsyncAPIDocument, originalAsyncAPI: "", params: {} });
        expect((content as TemplateRenderResult)?.content).toBe("hello Test"); 
      });
    });
  });
  describe('should transpile ES5 files', () => {
    describe('with a simple setup', () => {
      const es5_testFile = path.resolve(outputFiles, './ES5/simple.js');
      const es5_testFileMap = path.resolve(outputFiles, './ES5/simple.js.map');
      test('and import correctly', async () => {
        const content = await readFile(es5_testFile, 'utf8')
        expect(switchToUnixLinebreaks(content)).toMatchSnapshot();
        const mapContent = await readFile(es5_testFileMap, 'utf8');
        expect(switchToUnixLinebreaks(mapContent)).toMatchSnapshot();
        expect(await import(es5_testFile)).toBeDefined();
      });
      test('and render correctly', async () => {
        const content = await renderTemplate(es5_testFile, { asyncapi: {} as AsyncAPIDocument, originalAsyncAPI: "", params: {} });
        expect((content as TemplateRenderResult)?.content).toBe("hello Test"); 
      });
    });
  });
  describe('should transpile ES6 files', () => {
    describe('with a simple setup', () => {
      const es6_testFile = path.resolve(outputFiles, './ES6/simple.js');
      const es6_testFileMap = path.resolve(outputFiles, './ES6/simple.js.map');
      test('and import correctly', async () => {
        const content = await readFile(es6_testFile, 'utf8')
        expect(switchToUnixLinebreaks(content)).toMatchSnapshot();
        const mapContent = await readFile(es6_testFileMap, 'utf8');
        expect(switchToUnixLinebreaks(mapContent)).toMatchSnapshot();
        expect(await import(es6_testFile)).toBeDefined();
      });
      test('and render correctly', async () => {
        const content = await renderTemplate(es6_testFile, { asyncapi: {} as AsyncAPIDocument, originalAsyncAPI: "", params: {} });
        expect((content as TemplateRenderResult)?.content).toBe("hello Test"); 
      });
    });
  });
});

/*
  It is a helper required for snapshot testing on windows. It can't be solved by editor configuration and the end line setting because snapshots are generated not created in the editor.
  We need to remove `\r` from files transpiled on windows before we can match them with the snapshot generated on unix
*/
function switchToUnixLinebreaks(str: String) {
  return str.replace(/\\r/g, "")
}