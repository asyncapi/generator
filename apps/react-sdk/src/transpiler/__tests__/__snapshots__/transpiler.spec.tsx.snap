// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transpiler should transpile CommonJS files with a simple setup and import correctly 1`] = `
"'use strict';

var jsxRuntime = require('react/cjs/react-jsx-runtime.production.min');
require('source-map-support/register');

/* eslint-disable no-undef */
const path = require('path'); // this weird import are only necessary because we test within the SDK itself.
// eslint-disable-next-line security/detect-non-literal-require


const {
  File
} = require(path.resolve(__dirname, '../../../../components'));

function greetings(name) {
  return \`hello \${name}\`;
} // eslint-disable-next-line react/display-name


module.exports = function () {
  return /*#__PURE__*/jsxRuntime.jsx(File, {
    children: greetings('Test')
  });
};
//# sourceMappingURL=simple.js.map
"
`;

exports[`Transpiler should transpile CommonJS files with a simple setup and import correctly 2`] = `"{\\"version\\":3,\\"file\\":\\"simple.js\\",\\"sources\\":[\\"../../testfiles/CommonJS/simple.js\\"],\\"sourcesContent\\":[\\"/* eslint-disable no-undef */\\\\nconst path = require('path');\\\\n// this weird import are only necessary because we test within the SDK itself.\\\\n// eslint-disable-next-line security/detect-non-literal-require\\\\nconst {File} = require(path.resolve(__dirname, '../../../../components'));\\\\nfunction greetings(name) {\\\\n  return \`hello \${name}\`;\\\\n}\\\\n// eslint-disable-next-line react/display-name\\\\nmodule.exports = function() {\\\\n  return (\\\\n    <File>\\\\n      {greetings('Test')}\\\\n    </File>\\\\n  );\\\\n};\\"],\\"names\\":[\\"path\\",\\"require\\",\\"File\\",\\"resolve\\",\\"__dirname\\",\\"greetings\\",\\"name\\",\\"module\\",\\"exports\\",\\"_jsx\\"],\\"mappings\\":\\";;;;;AAAA;AACA,MAAMA,IAAI,GAAGC,OAAO,CAAC,MAAD,CAApB;AAEA;;;AACA,MAAM;AAACC,EAAAA;AAAD,IAASD,OAAO,CAACD,IAAI,CAACG,OAAL,CAAaC,SAAb,EAAwB,wBAAxB,CAAD,CAAtB;;AACA,SAASC,SAAT,CAAmBC,IAAnB,EAAyB;AACvB,SAAQ,SAAQA,IAAK,EAArB;AACD;;;AAEDC,MAAM,CAACC,OAAP,GAAiB,YAAW;AAC1B,sBACEC,eAAC,IAAD;AAAA,cACGJ,SAAS,CAAC,MAAD;AADZ,IADF;AAKD,CAND;;\\"}"`;

exports[`Transpiler should transpile ES5 files with a simple setup and import correctly 1`] = `
"'use strict';

var jsxRuntime = require('react/cjs/react-jsx-runtime.production.min');
require('source-map-support/register');
var path = require('path');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var path__default = /*#__PURE__*/_interopDefaultLegacy(path);

// eslint-disable-next-line security/detect-non-literal-require, no-undef

const {
  File
} = require(path__default['default'].resolve(__dirname, '../../../../components'));

function greetings(name) {
  return \`hello \${name}\`;
} // eslint-disable-next-line react/display-name


function simple () {
  return /*#__PURE__*/jsxRuntime.jsx(File, {
    children: greetings('Test')
  });
}

module.exports = simple;
//# sourceMappingURL=simple.js.map
"
`;

exports[`Transpiler should transpile ES5 files with a simple setup and import correctly 2`] = `"{\\"version\\":3,\\"file\\":\\"simple.js\\",\\"sources\\":[\\"../../testfiles/ES5/simple.js\\"],\\"sourcesContent\\":[\\"import path from 'path';\\\\n// this weird import are only necessary because we test within the SDK itself.\\\\n// eslint-disable-next-line security/detect-non-literal-require, no-undef\\\\nconst {File} = require(path.resolve(__dirname, '../../../../components'));\\\\nfunction greetings(name) {\\\\n  return \`hello \${name}\`;\\\\n}\\\\n// eslint-disable-next-line react/display-name\\\\nexport default function() {\\\\n  return (\\\\n    <File>\\\\n      {greetings('Test')}\\\\n    </File>\\\\n  );\\\\n}\\"],\\"names\\":[\\"File\\",\\"require\\",\\"path\\",\\"resolve\\",\\"__dirname\\",\\"greetings\\",\\"name\\",\\"_jsx\\"],\\"mappings\\":\\";;;;;;;;;;AAEA;;AACA,MAAM;AAACA,EAAAA;AAAD,IAASC,OAAO,CAACC,wBAAI,CAACC,OAAL,CAAaC,SAAb,EAAwB,wBAAxB,CAAD,CAAtB;;AACA,SAASC,SAAT,CAAmBC,IAAnB,EAAyB;AACvB,SAAQ,SAAQA,IAAK,EAArB;AACD;;;AAEc,mBAAW;AACxB,sBACEC,eAAC,IAAD;AAAA,cACGF,SAAS,CAAC,MAAD;AADZ,IADF;AAKD;;;;\\"}"`;

exports[`Transpiler should transpile ES6 files with a simple setup and import correctly 1`] = `
"'use strict';

var jsxRuntime = require('react/cjs/react-jsx-runtime.production.min');
require('source-map-support/register');
var path = require('path');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var path__default = /*#__PURE__*/_interopDefaultLegacy(path);

// eslint-disable-next-line security/detect-non-literal-require, no-undef

const {
  File
} = require(path__default['default'].resolve(__dirname, '../../../../components'));

const greetings = name => \`hello \${name}\`; // eslint-disable-next-line react/display-name


function simple () {
  return /*#__PURE__*/jsxRuntime.jsx(File, {
    children: greetings('Test')
  });
}

module.exports = simple;
//# sourceMappingURL=simple.js.map
"
`;

exports[`Transpiler should transpile ES6 files with a simple setup and import correctly 2`] = `"{\\"version\\":3,\\"file\\":\\"simple.js\\",\\"sources\\":[\\"../../testfiles/ES6/simple.js\\"],\\"sourcesContent\\":[\\"import path from 'path';\\\\n// this weird import are only necessary because we test within the SDK itself.\\\\n// eslint-disable-next-line security/detect-non-literal-require, no-undef\\\\nconst {File} = require(path.resolve(__dirname, '../../../../components'));\\\\nconst greetings = name => \`hello \${name}\`;\\\\n// eslint-disable-next-line react/display-name\\\\nexport default function() {\\\\n  return (\\\\n    <File>\\\\n      {greetings('Test')}\\\\n    </File>\\\\n  );\\\\n}\\"],\\"names\\":[\\"File\\",\\"require\\",\\"path\\",\\"resolve\\",\\"__dirname\\",\\"greetings\\",\\"name\\",\\"_jsx\\"],\\"mappings\\":\\";;;;;;;;;;AAEA;;AACA,MAAM;AAACA,EAAAA;AAAD,IAASC,OAAO,CAACC,wBAAI,CAACC,OAAL,CAAaC,SAAb,EAAwB,wBAAxB,CAAD,CAAtB;;AACA,MAAMC,SAAS,GAAGC,IAAI,IAAK,SAAQA,IAAK,EAAxC;;;AAEe,mBAAW;AACxB,sBACEC,eAAC,IAAD;AAAA,cACGF,SAAS,CAAC,MAAD;AADZ,IADF;AAKD;;;;\\"}"`;
